tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -4721,6 +4721,58 @@ Status ConvertSoftmax(OpConverterParams* params) {
   return Status::OK();
 }
 
+Status ConvertLogSoftmax(OpConverterParams* params) {
+  const auto& inputs = params->inputs;
+  const auto& node_def = params->node_def;
+  TF_RETURN_IF_ERROR(CheckInputsWeights(*params, {{"logits", false}}));
+  TF_RETURN_IF_ERROR(
+      AllowDataTypes(*params, {DataType::DT_FLOAT, DataType::DT_HALF}));
+
+  ITensorProxyPtr logits_tensor = inputs.at(0).tensor();
+
+  const int num_trt_dims = logits_tensor->getDimensions().nbDims;
+  if (!num_trt_dims && params->use_implicit_batch) {
+    return errors::InvalidArgument(
+        "TensorRT LogSoftmax cannot apply on the batch dimension");
+  }
+
+  if (params->validation_only) return Status::OK();
+
+  // Perform LogSoftmax operation -
+  // logsoftmax = logits - log(reduce_sum(exp(logits), axis))
```

Comment:
Is there a reason to prefer this formulation over  `log(softmax(logits))`?

Reply:
I chose this way because tensorflow implements in Python using `logsoftmax = logits - log(reduce_sum(exp(logits), axis))` .